homeassistant/components/ecowitt/entity.py

```diff
@@ -21,11 +21,12 @@ def __init__(self, sensor: EcoWittSensor) -> None:
         self.ecowitt: EcoWittSensor = sensor
 
         self._attr_unique_id = f"{sensor.station.key}-{sensor.key}"
+        device_name = sensor.station.station.split("_")[0]
```

Comment:
I think we should make sure we only remove the last bit...

If I have named my device "garden_backyard", this approach becomes problematic.

Reply:
afaik this name can't be changed

Reply:
The other option would be to use the model as name (as in the issue multiple people report that is the expected name). But I chose this approach because there probably was a reason to use this field in the first place

Reply:
I think @frenck meant splitting is fine - if we know the station is alwayse suffixed with _<version> - but we should _drop the last item_ instead of _picking the first_.
So, something like this:
```
        device_name = sensor.station.station.rsplit("_", 1)[0]
```